### PR TITLE
Implement Led command

### PR DESCRIPTION
**Pull request by skywhale**

### CODE REVIEW

File: src/rgb_led.rs [hunk 1]
```diff
@@ -76,3 +78,27 @@ impl<F: FullDuplex<u8>> LedStrip<F> {
...
+
+    #[allow(dead_code)]
+    pub fn intensity(&self) -> f32 {
+        let intervals = self.instant.elapsed() as f32 / self.interval_ticks;
+        let pulse = (libm::sinf(intervals) + 1.0) * 0.5;
+        let skip_one = if libm::sinf((intervals + PI / 2.0) / 2.0) >= 0.0 { 1.0 } else { 0.0 };
+
+        pulse * skip_one
+    }
+}
```
Comment (bschwind) [inline src/rgb_led.rs:104]: This is much better! I changed the pulse interval to 700ms and it feels pretty close to the old Mac sleep LED.

I do notice a slight "skip" on occasion when it's increasing in brightness, not yet sure what that's about. Possibly when the mono timer's elapsed time reaches `u32::max`?

Comment (bschwind) [reply src/rgb_led.rs:104]: So far I can confirm that it happens after 1 minute, 30 seconds into the firmware starting up, which lines up with a 48MHz clock.

(2^32) / 48,000,000 = 89.48 ~= 1 minute, 29 seconds

Comment (skywhale) [reply src/rgb_led.rs]: @bschwind good catch :) I don't know how the u32 counter wraps, but do you mind testing my latest commit which tries to work it around?

Comment (skywhale) [reply src/rgb_led.rs]: They seem to wrap to zero when overflowing, so maybe my code works:
https://stackoverflow.com/questions/53434504/stm32f411-dwt-cyccnt-counter-width

File: src/rgb_led.rs [hunk 2]
```diff
@@ -76,3 +78,58 @@ impl<F: FullDuplex<u8>> LedStrip<F> {
...
+            elapsed: elapsed as u64,
+            last_elapsed: elapsed,
+            instant, 
+        }
+    }
+}
+
+impl U64Instant {
+    fn elapsed(&mut self) -> u64 {
+        let mut diff = self.instant.elapsed() as i32 - self.last_elapsed as i32;
```
Comment (bschwind) [inline src/rgb_led.rs:105]: Does it make sense here to cast a `u32` as an `i32`?

File: src/rgb_led.rs [hunk 3]
```diff
@@ -76,3 +78,58 @@ impl<F: FullDuplex<u8>> LedStrip<F> {
...
+        if diff < 0 {
+            diff += u32::MAX as i32 + 1;
```
Comment (bschwind) [inline src/rgb_led.rs:107]: `u32::MAX as i32` doesn't that always evaluate to `-1`?

Comment (skywhale) [reply src/rgb_led.rs:107]: oh yeah it does...

File: src/rgb_led.rs (hunk 2 above)
Comment (skywhale) [reply src/rgb_led.rs:105]: it doesn't... thanks for the catch!!

**Comment by skywhale** [comment]
I was missing a line to set `last_elapsed`. Fixed and tested. The sudden change with the brightness after 1.5 mins is goine.

**Pull request merged**